src/api/v2/utils.js

```diff
@@ -43,10 +43,25 @@ export function isIp(hostname) {
  * @throws {InvalidHostNameIpError | InvalidHostNameError}
  */
 export async function validHostname(hostname) {
-  // No hostnames lacking a dot or something like localhost
+  // remove any trailing dot
+  hostname = hostname.replace(/\.$/, "");
   if (
     !hostname.includes(".") ||
-    hostname.includes("localhost") ||
+    hostname === "localhost" ||
+    // RFC 2606
+    hostname.endsWith(".test") ||
+    hostname.endsWith(".example") ||
+    hostname.endsWith(".invalid") ||
+    hostname.endsWith(".localhost") ||
+    // RFC 6761
+    hostname === "example.com" ||
+    hostname.endsWith(".example.com") ||
+    hostname === "example.net" ||
+    hostname.endsWith(".example.net") ||
+    hostname === "example.org" ||
+    hostname.endsWith(".example.org") ||
```

Comment:
I'd keep these in (at least the top level `example.{com, net, org}`) as they are valid domains, and useful for scanning an example site to ensure everything's working.

Reply:
Ok, agreed.